src/Sulu/Bundle/MediaBundle/Media/Storage/S3Storage.php

```diff
@@ -44,14 +49,16 @@ public function __construct(FilesystemInterface $filesystem, int $segments)
 
         $this->endpoint = (string) $this->adapter->getClient()->getEndpoint();
         $this->bucketName = $this->adapter->getBucket();
+
+        $this->url = null !== $url ? $url : $this->endpoint . '/' . $this->bucketName;
```

Comment:
is the bucketName really not part of the url? https://3v4l.org/8JNSO

Reply:
That is correct, the CDN doesn't have the bucket name in the folder structure.

Digitalocean CDN: https://bucketname.fra1.cdn.digitaloceanspaces.com/file.example
Cloudflare: https://customdomain.com/file.example

It is important to not automatically add the bucketName to the url.